src/request/mod.rs

```diff
@@ -64,7 +64,8 @@ impl<'a> Request<'a> {
     /// Create a request from an HttpRequest.
     ///
     /// This constructor consumes the HttpRequest.
-    pub fn from_http(req: HttpRequest<'a>, local_addr: SocketAddr) -> Result<Request<'a>, String> {
+    pub fn from_http(req: HttpRequest<'a>, local_addr: SocketAddr, protocol: &str)
```

Comment:
If we changed `Security` to `Protocol` then the `protocol` argument here could be changed to `Protocol` from `&str`.